cmd/stat-main.go

```diff
@@ -0,0 +1,153 @@
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+// stat specific flags.
+var (
+	statFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "recursive, r",
+			Usage: "Stat recursively.",
+		},
+	}
+)
+
+// stat contents of files and folders.
+var statCmd = cli.Command{
+	Name:   "stat",
+	Usage:  "Stat contents of objects",
+	Action: mainStat,
+	Before: setGlobalsFromContext,
+	Flags:  append(statFlags, globalFlags...),
+	CustomHelpTemplate: `NAME:
+	 {{.HelpName}} - {{.Usage}}
+ 
+ USAGE:
+	 {{.HelpName}} [FLAGS] TARGET [TARGET ...]
+ 
+ FLAGS:
+	 {{range .VisibleFlags}}{{.}}
+	 {{end}}
+ EXAMPLES:
+		1. Stat all contents of mybucket on Amazon S3 cloud storage.
+			 $ {{.HelpName}} s3/mybucket/
+ 
+		2. Stat all contents of mybucket on Amazon S3 cloud storage on Microsoft Windows.
+			 $ {{.HelpName}} s3\mybucket\
+ 
+		3. Stat files recursively on a local filesystem on Microsoft Windows.
+			 $ {{.HelpName}} --recursive C:\Users\Worf\
+ `,
+}
+
+// checkStatSyntax - validate all the passed arguments
+func checkStatSyntax(ctx *cli.Context) {
+	args := ctx.Args()
+	if !ctx.Args().Present() {
+		args = []string{"."}
+	}
+	for _, arg := range args {
+		if strings.TrimSpace(arg) == "" {
+			fatalIf(errInvalidArgument().Trace(args...), "Unable to validate empty argument.")
+		}
+	}
+	// extract URLs.
+	URLs := ctx.Args()
+	isIncomplete := false
+
+	for _, url := range URLs {
+		_, _, err := url2Stat(url)
+		if err != nil && !isURLPrefixExists(url, isIncomplete) {
+			// Bucket name empty is a valid error for 'ls myminio',
+			// treat it as such.
+			if _, ok := err.ToGoError().(BucketNameEmpty); ok {
+				continue
+			}
+			fatalIf(err.Trace(url), "Unable to stat `"+url+"`.")
+		}
+	}
+}
+
+// mainStat - is a handler for mc stat command
+func mainStat(ctx *cli.Context) error {
+	// Additional command specific theme customization.
+	console.SetColor("Name", color.New(color.Bold, color.FgCyan))
+	console.SetColor("Date", color.New(color.FgWhite))
+	console.SetColor("Size", color.New(color.FgWhite))
+	console.SetColor("ETag", color.New(color.FgWhite))
+
+	console.SetColor("EncryptionHeaders", color.New(color.FgWhite))
+	console.SetColor("Metadata", color.New(color.FgWhite))
+
+	// check 'stat' cli arguments.
+	checkStatSyntax(ctx)
+
+	// Set command flags from context.
+	isRecursive := ctx.Bool("recursive")
+
+	args := ctx.Args()
+	// mimic operating system tool behavior.
+	if !ctx.Args().Present() {
+		args = []string{"."}
+	}
+
+	var cErr error
+	for _, targetURL := range args {
+		var clnt Client
+		clnt, err := newClient(targetURL)
+		fatalIf(err.Trace(targetURL), "Unable to initialize target `"+targetURL+"`.")
+		if isAliasURLDir(targetURL) && !isRecursive {
+			fatalIf(errInvalidArgument(), "Use --recursive option to stat by prefix")
+
+		}
+		if !isRecursive || !isAliasURLDir(targetURL) {
+			var st *clientContent
```

Comment:
I recommend that you move this block in doStat() itself so it will be more easier to read

Reply:
done.